### PR TITLE
feat: Add FaceIterator for external access to faces in ConvexPolyhedron

### PR DESCRIPTION
**Pull request by Rhuagh**
Also ran rustfmt-nightly.

This change is [Reviewable](https://reviewable.io/reviews/kvark/collision-rs/55)

### CODE REVIEW

File: src/primitive/polyhedron.rs [hunk 1]
```diff
@@ -169,6 +175,49 @@ where
...
+    S: BaseFloat,
+{
+    type Item = (&'a Point3<S>, &'a Point3<S>, &'a Point3<S>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current >= self.polyhedron.faces.len() {
+            None
+        } else {
+            self.current += 1;
+            let face = &self.polyhedron.faces[self.current - 1];
```
Comment (kvark) [inline src/primitive/polyhedron.rs]: would be simpler to get a face first, and then increase the current id

File: src/primitive/polyhedron.rs [hunk 2]
```diff
@@ -169,6 +175,49 @@ where
...
+{
+    polyhedron: &'a ConvexPolyhedron<S>,
+    current: usize,
+}
+
+impl<'a, S: 'a> FaceIterator<'a, S>
+where
+    S: BaseFloat,
+{
+    pub fn new(polyhedron: &'a ConvexPolyhedron<S>) -> Self {
```
Comment (kvark) [inline src/primitive/polyhedron.rs]: do we actually need this constructor exposed to the user? and if not, we can get away without the function at all

**Comment by Rhuagh** [comment]
It is rustfmt that did all the style changes. I always run the newest version before pushing.

File: src/primitive/polyhedron.rs (hunk 2 above)
Comment (Rhuagh) [reply src/primitive/polyhedron.rs]: True.

File: src/primitive/polyhedron.rs (hunk 1 above)
Comment (Rhuagh) [reply src/primitive/polyhedron.rs]: Possibly.

**Comment by kvark** [comment]
hmm, ok.
This is rather unexpected, but on the other side - it's good, since this is how I write too :)
bors r+

**Comment by Rhuagh** [comment]
Been a while since we did a formatting pass, probably why there were so many changes.

**Pull request merged**